clang/lib/AST/Expr.cpp

```diff
@@ -648,6 +649,85 @@ UniqueStableNameDiscriminator(ASTContext &, const NamedDecl *ND) {
   return llvm::None;
 }
 
+/// Compute a unique name that is consumable by sycl_vxx
+std::string computeUniqueSYCLVXXName(StringRef Demangle, bool shouldRename) {
+  if (!shouldRename)
+    return Demangle.str();
+
+  /// VXX has a maximum of 64 character for the name of the kernel function
+  /// plus the name of one parameter.
+  /// Those characters need to be used wisely to prevent name collisions.
+  /// It is also useful to use a name that is understandable by the user,
+  /// so we add only 8 character of hash and only if needed.
+  /// The first character cannot be an underscore or a digit.
+  /// An underscore can only be found between two non underscore character
+  /// (No __ or undersocre at end or start of name)
```

Comment:
```suggestion
  /// (No __ or underscore at end or start of name)
```
Spell checker in comments...